homework/shared_ptr/shared_ptr.hpp

```diff
@@ -102,7 +102,15 @@ const T* shared_ptr<T>::get() const {
 
 template <typename T>
 void shared_ptr<T>::reset(T* newPtr) {
-    shared_ptr{newPtr}.swap(*this);
+    if (counter_) {
+        if (counter_->getRefs() == 1) {
+            delete ptr_;
+        } else {
+            --(*counter_);
+        }
+    }
+    ptr_ = newPtr;
+    counter_ = new control_block();
```

Comment:
```suggestion
    if (counter_) {
        if (counter_->getRefs() == 1) {
            delete ptr_;
            //needs to be checked for weakRefs if counter is needed
        } else {
            --(*counter_);
        }
    }
    ptr_ = newPtr;
    counter_ = new control_block();
    ++*counter_;
    
```

Something along those lines